src/OrasProject.Oras/Pack.cs

```diff
@@ -0,0 +1,335 @@
+    /// References:
+    /// - https://github.com/opencontainers/image-spec/blob/v1.1.0/schema/defs-descriptor.json#L7
+    /// - https://datatracker.ietf.org/doc/html/rfc6838#section-4.2
+    /// ^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}/[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}$
+    /// </summary>
+    private const string _mediaTypeRegexp = @"^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}/[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}$";
+    /// <summary>
+    /// private static readonly Regex _mediaTypeRegex = new Regex(_mediaTypeRegexp, RegexOptions.Compiled);
+    /// </summary>
+    private static readonly Regex _mediaTypeRegex = new Regex(@"^[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}/[A-Za-z0-9][A-Za-z0-9!#$&-^_.+]{0,126}(\+json)?$", RegexOptions.Compiled);
+
+    /// <summary>
+    /// PackManifest generates an OCI Image Manifestbased on the given parameters
+    /// and pushes the packed manifest to a content storage/registry using pusher. The version
+    /// of the manifest to be packed is determined by packManifestVersion
+    /// (Recommended value: PackManifestVersion1_1).
+    /// - If packManifestVersion is [PackManifestVersion1_1]:
+    ///   artifactType MUST NOT be empty unless PackManifestOptions.ConfigDescriptor is specified.
+    /// - If packManifestVersion is [PackManifestVersion1_0]:
+    ///   if PackManifestOptions.ConfigDescriptor is null, artifactType will be used as the
+    ///   config media type; if artifactType is empty,
+    ///   "application/vnd.unknown.config.v1+json" will be used.
+    ///   if PackManifestOptions.ConfigDescriptor is NOT null, artifactType will be ignored.
+    ///
+    /// artifactType and PackManifestOptions.ConfigDescriptor.MediaType MUST comply with RFC 6838.
+    ///
+    /// Each time when PackManifest is called, if a time stamp is not specified, a new time
+    /// stamp is generated in the manifest annotations with the key ocispec.AnnotationCreated
+    /// (i.e. "org.opencontainers.image.created"). To make [PackManifest] reproducible,
+    /// set the key ocispec.AnnotationCreated to a fixed value in
+    /// opts.Annotations. The value MUST conform to RFC 3339.
+    ///
+    /// If succeeded, returns a descriptor of the packed manifest.
+    ///
+    /// Note: PackManifest can also pack artifact other than OCI image, but the config.mediaType value 
+    /// should not be a known OCI image config media type [PackManifestVersion1_1]
+    /// </summary>
+    /// <param name="pusher"></param>
+    /// <param name="version"></param>
+    /// <param name="artifactType"></param>
+    /// <param name="options"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <exception cref="NotSupportedException"></exception>
+    public static async Task<Descriptor> PackManifestAsync(
+            ITarget pusher,
+            PackManifestVersion version,
+            string? artifactType,
+            PackManifestOptions options,

```

Comment:
nit: will this work?
```suggestion
            PackManifestOptions options = default,

```

Reply:
yes, changed